homeassistant/components/tplink/light.py

```diff
@@ -60,20 +61,12 @@
 
 async def async_setup_entry(hass: HomeAssistantType, config_entry, async_add_entities):
     """Set up lights."""
-    devices = hass.data[TPLINK_DOMAIN][CONF_LIGHT]
-    entities = []
+    await hass.async_add_executor_job(
+        add_available_devices, hass, CONF_LIGHT, TPLinkSmartBulb, async_add_entities
```

Comment:
We're using `async_add_entities` inside the executor thread pool. That's not ok. The async api needs to be run inside the event loop.

Reply:
Sorry about that, submitted new PR: https://github.com/home-assistant/core/pull/42454